File: articles.jdbc/src/test/java/pet/db/jdbc/tool/producer/AuthenticationDetailsProducer.java
package pet.db.jdbc.tool.producer;

import jakarta.validation.constraints.NotNull;

import org.springframework.security.core.userdetails.UserDetails;

import pet.db.jdbc.model.dto.User;
import pet.db.jdbc.model.enums.UserRole;

public interface AuthenticationDetailsProducer {

    UserDetails produceUserDetailsOfRegisteredUser(@NotNull UserRole role);

    User produceRegisteredUserWithRawPassword(@NotNull UserRole role);

}

File: src/test/kotlin/pet/articles/tool/producer/AuthenticationDetailsProducer.kt
package pet.articles.tool.producer


import pet.articles.model.dto.User
import pet.articles.model.enums.UserRole

interface AuthenticationDetailsProducer {

    fun produceRegisteredUserWithRawPassword(role: UserRole = UserRole.ROLE_USER): User

    fun produceRegisteredUser(role: UserRole = UserRole.ROLE_USER): User
}